File: backend/app/schemas/google_drive.py
"""Request/response schemas for the Google Drive OAuth connection."""

from datetime import datetime

from pydantic import BaseModel


class GoogleDriveConnectionStatus(BaseModel):
    connected: bool
    google_email: str | None = None
    connected_at: datetime | None = None


class GoogleDriveConnectAuthorizationUrl(BaseModel):
    authorization_url: str

File: frontend/src/lib/api/googleDrive.ts
import { apiFetch } from "./client";

export interface GoogleDriveConnectionStatus {
  connected: boolean;
  google_email: string | null;
  connected_at: string | null;
}

export function getGoogleDriveConnectionStatus(
  token: string,
  signal?: AbortSignal,
): Promise<GoogleDriveConnectionStatus> {
  return apiFetch<GoogleDriveConnectionStatus>("/google-drive/connection", { token, signal });
}

export function getGoogleDriveConnectAuthorizationUrl(
  token: string,
): Promise<{ authorization_url: string }> {
  return apiFetch<{ authorization_url: string }>("/google-drive/connect", { token });
}

export function disconnectGoogleDrive(token: string): Promise<undefined> {
  return apiFetch<undefined>("/google-drive/connection", { method: "DELETE", token });
}
